### PR TITLE
Fix the Week Of / Month Of logic

### LINKED ISSUE #5939: "Week of" schedule
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

Currently, all weeks types (at least on the first month, as I've tested 50+ games) on all maps happen completely the same.
After the first week we always see the message "Week of..." rising growth of particular creature +5. Creature is always different, that works as intended. 👍 
There're common week messages after the 2nd and 3rd weeks when nothing is happened.
There's always doubled creature growth at the end of the first month.
This behaviour is completely the same every map we start.
In the OG common weeks and "weeks of ... +5" happen randomly (75% and 25% accordingly)
In the OG months appear like:
- 50% common month without extra creatures growth
- 40% a month with double creature growth
- 10% a month of PLAGUE.

Currently, some common months tend to occasionally appear as plague, but the first month, which always has doubled creatures growth - never.
image

### Save file

Just start any map.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #5939

### CODE REVIEW

File: src/fheroes2/kingdom/week.h [hunk 1]
```diff
@@ -29,34 +29,42 @@
...
     SQUIRREL,
     RABBIT,
     GOPHER,
     BADGER,
+    RAT,
     EAGLE,
     WEASEL,
     RAVEN,
     MONGOOSE,
+    HOUND,
```
[the remark this answers is not in the document]
Comment (zenseii) [reply src/fheroes2/kingdom/week.h:43]: We will have to check this somewhere, but I think hound sounds better in the context of this medieval fantasy game than "Dog" which sounds a bit more basic? Like Hell hounds and not hell dogs.

[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/kingdom/week.h:43]: @zenseii The problem is that this "Week Of" was missing before this PR, and I have its name only in Russian from this article:

https://handbookhmm.ru/2-timing-and-garrison

In Russian this week is named as "Собаки", which can be translated as "Dog" or "Hound". I cannot find how it was named in the original game in English.

[the remark this answers is not in the document]
Comment (zenseii) [reply src/fheroes2/kingdom/week.h:43]: @oleg-derevenetz It's seem like dog it is. This is the original game (only took 12 months and 4 weeks to get there...):
image

[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/kingdom/week.h]: @zenseii Okay, it's a "Dog" after all, not "Hound". Thank you very much!

[the remark this answers is not in the document]
Comment (zenseii) [reply src/fheroes2/kingdom/week.h]: You're welcome!

**Comment by ihhub** [comment]
@oleg-derevenetz , huge thanks for this fix!

**Pull request merged**